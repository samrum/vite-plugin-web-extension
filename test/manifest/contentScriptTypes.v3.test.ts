import { getResourceDir, runManifestV3Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("contentScriptTypes");

runManifestV3Test("contentScriptTypes", () => ({
  content_scripts: [
    {
      js: [`${resourceDir}/content1.js`],
      matches: ["https://example.com/"],
    },
    {
      js: [`${resourceDir}/content2.ts`],
      matches: ["https://example.com/"],
    },
  ],
}));
