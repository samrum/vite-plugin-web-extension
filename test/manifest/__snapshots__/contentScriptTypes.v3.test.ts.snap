// Vitest Snapshot v1

exports[`contentScriptTypes - Manifest V3 1`] = `
[
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  \\"version\\": \\"1.0.0\\",
  \\"name\\": \\"Manifest Name\\",
  \\"manifest_version\\": 3,
  \\"content_scripts\\": [
    {
      \\"js\\": [
        \\"test/manifest/resources/contentScriptTypes/content1.js\\"
      ],
      \\"matches\\": [
        \\"https://example.com/\\"
      ]
    },
    {
      \\"js\\": [
        \\"test/manifest/resources/contentScriptTypes/content2.js\\"
      ],
      \\"matches\\": [
        \\"https://example.com/\\"
      ]
    }
  ]
}",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/contentScriptTypes/content1.js",
    "name": undefined,
    "source": "console.log(\\"content1\\");
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/contentScriptTypes/content2.js",
    "name": undefined,
    "source": "console.log(\\"content2\\");
",
    "type": "asset",
  },
]
`;
